modules/model/item/trait.js

```diff
@@ -164,4 +164,10 @@ export class TraitModel extends PropertiesMixin(BaseItemModel)
       }
 
 
+    shouldTransferEffect(effect)
+    {
+      let isExcluded = this.parent.actor?.excludedTraits?.includes(this.parent._id) ?? false;
+      return isExcluded === false;
+    }
```

Comment:
change this to `return !this.parent.actor?.excludedTraits?.includes(this.id)` and I will merge.